### PR TITLE
Moving MySQL driver to use autocommit

### LINKED ISSUE #7234: [mysql-integration] Enable autocommit
### Short description and motivation for the proposed feature

Currently, when a MySQL query succeeds a commit is made explicitly, and when a query fails a rollback is made. Autocommit is not disabled explicitly, but it's disabled by default by all Python MySQL modules.

Using autocommit is better in principle, and it will avoid that a transaction remains open if a Python error occurs.

### Video or screenshots

_No response_

### Describe some possible solutions

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by einhverfr**
This changes the MySQL integration to use autocommit. This fixes #7234. The rationale in that ticket is that python errors, if caught, could lead to dangling open transactions with unexpected results.  This concern is valid.  If we need transactions we should provide a context manager for them so that exiting the context manager automaticaly closes the transaction.

Due to the nature of this change, there are no extra integration tests required.  Nor is there any additional documentation needed aside from something in the module docstring which I have already added.

## Description
In practical terms, this prevents dangling transactions in the case of unexpected python errors and therefore makes the module more robust.

Fixes #7234 

## Type of change

(Please delete options that are not relevant)

- [X] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [X]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.
 
I added an integration test that makes sure that the data which is written persists.  This disconnects and then reconnects in the middle of the test.

The changes here are such that existing MySQL integration tests should test the functionality properly.  I modified one integration test to ensure that writes persist 

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [X] Necessary documentation updates are either made or tracked in issues.
- [X] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: tests/handler_tests/test_mysql_handler.py [hunk 1]
```diff
@@ -136,8 +136,20 @@ def test_create_table(self, handler):
...
+        handler.disconnect()
+        handler.connect()
+        res = handler.query(f"SELECT count(*) FROM {new_table}")
+        self.check_valid_response(res)
+        got_rows = res.data_frame.shape[0]
+        assert got_rows == 3
+
+
     def test_drop_table(self, handler):
-        drop_table = "test_md"
```
Comment (einhverfr) [inline tests/handler_tests/test_mysql_handler.py:140]: I am guessing that this is a typo?

Comment (ZoranPandovski) [reply tests/handler_tests/test_mysql_handler.py:140]: It looks so, anyway we will need to change this tests

**Review by ZoranPandovski — APPROVED** [review]

**Comment by ZoranPandovski** [comment]
@einhverfr While you are working on this PR can you please help and solve the flake8 warnings https://github.com/mindsdb/mindsdb/actions/runs/7000786901/job/19364274897?pr=8429#step:6:115

**Comment by ZoranPandovski** [comment]
@einhverfr Can you please solve the flake8 warnings https://github.com/mindsdb/mindsdb/actions/runs/7000786901/job/19364274897?pr=8429#step:6:115

**Pull request merged**